File: 539. Minimum Time Difference.py
class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:
        ts = sorted([self.timetoint(time) for time in timePoints])
        mindiff = 2**31-1
        for i, tm in enumerate(ts):
            if i < len(ts)-1:
                mindiff = min(mindiff, ts[i+1]-tm)
            else:
                diff = 23*60+59-tm + 1 + ts[0]
                mindiff = min(mindiff, diff)
        return mindiff
        
    def timetoint(self, time):
        t = time.split(':')
        return int(t[0])*60+int(t[1])

File: 539. Minimum Time Difference.java
class Solution {
    public int findMinDifference(List<String> timePoints) {
        boolean[] exist = new boolean[24*60];
        for (String s : timePoints) {
            String[] sp = s.split(":");
            int temp = Integer.parseInt(sp[0])*60 + Integer.parseInt(sp[1]);
            if (exist[temp]) {
                return 0;
            }
            exist[temp] = true;
        }
        int ans = 24*60;
        int prev = -1;
        for (int i=0; i<24*60; i++) {
            if (exist[i]) {
                if (prev >= 0) {
                    ans = Math.min(ans, i-prev);
                    prev = i;
                } else {
                    prev = i;
                }
            }
        }
        for (int i=0; i<24*60; i++) {
            if (exist[i]) {
                ans = Math.min(ans, i+24*60-prev);
            }
        }
        return ans;
    }
}